File: raw/count_bits.py
import time


def countSetBits(n):
    count = 0
    while n:
        count += n & 1
        n >>= 1
    return count


def count_set_bits(n):
    total = 0
    while n:
        n &= n - 1
        total += 1
    return total


def count_bits(n):
    return bin(n)[2:].count('1')


def timeit(f):
    t1 = time.time()
    for i in range(1, 4000000):
        f(i)
    t2 = time.time()
    print(f'{f.__name__} took {round(t2-t1, 7)} seconds.')


if __name__ == '__main__':
    timeit(count_bits)
    timeit(countSetBits)
    timeit(count_set_bits)

File: raw/count_bits.cpp
#include <iostream>

unsigned long long count_bits(unsigned long long n) {
	unsigned long long total = 0;
	while (n > 0) {
		n &= n - 1;
		total += 1;
	}
	return total;
}

int main() {
	std::cout << count_bits(5) << std::endl;
	std::cout << count_bits(100) << std::endl;
	return 0;
}
